crates/solvers/src/tests/baseline/direct_swap.rs

```diff
@@ -0,0 +1,105 @@
+//! Simple test case that verifies that the baseline solver can settle an order
+//! directly with a Uniswap V2 pool.
+
+use serde_json::json;
+use crate::{cli, tests};
+use std::num::NonZeroUsize;
+
+#[tokio::test]
+async fn test() {
+    let engine = tests::SolverEngine::new(cli::Command::Baseline(cli::baseline::Arguments {
+        weth: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
+            .parse()
+            .unwrap(),
+        base_tokens: Vec::new(),
+        max_hops: NonZeroUsize::new(2).unwrap(),
```

Comment:
Wasn't `max_hops == 1` supposed to be used for direct swaps?

Reply:
We could test that `max_hops` gets respected as an upper limit by introducing pool data which would lead to a better settlement with 2 hops.

Reply:
For sure! That should be in a separate E2E test though right?

Reply:
I have no problem with adding that in another PR. Just wanted to mention it because this would have caught the issue I pointed out.

Reply:
Started writing the E2E test for this, and I found a bug 🙈. Hops are **additional tokens** in a path, and not **additional segments**. So, by the baseline solver code, `A -> B` is a 0-hop path, `A -> B -> C` is a 1 hop path, etc.